src/bin/main.rs

```diff
@@ -50,12 +57,57 @@ fn core() -> Result<i32, String> {
                     _ => None,
                 },
             )?;
+
+            println!("    Done!");
+
             return Ok(0);
         } else if prog_args[0] == "-h" || prog_args[0] == "--help" {
             help(stdout);
             return Ok(0);
         } else if prog_args[0] == "-c" || prog_args[0] == "--cache" {
             should_cache = true;
+        } else if prog_args[0] == "-e" || prog_args[0] == "--edit-template" {
+            let template_path: String = match get_template_path() {
+                Ok(path) => Ok(path.to_str().unwrap().to_string()),
+                Err(err) => Err(format!(
+                    "Failed to get template path with the following error: {:#?}",
+                    err
+                )),
+            }?;
+
+            let child;
+
+            if cfg!(target_os = "windows") {
+                // We need to spawn a `powershell` process to make `start` available.
+                child = OsCommand::new("powershell")
+                    .arg(format!("start '{}'", template_path))
+                    .spawn()
+                    .map(|mut x| x.wait());
+            } else {
+                let editor = PathBuf::from(env::var("EDITOR").unwrap_or("nano".to_string()));
+
+                let safe_editor = editor.to_str().ok_or(
+                    "Failed to parse the value within the `EDITOR` environment variable as a path/",
+                )?;
+
+                child = OsCommand::new(safe_editor)
+                    .arg(template_path)
+                    .spawn()
+                    .map(|mut x| x.wait());
```

Comment:
Have you documented that any given editor is expected to take the path as its first and only argument? I think this is a reasonable assumption, but it should be documented.